### PR TITLE
Feat: Adds `ProductCard` component

### PR DESCRIPTION
**Pull request by renatamottam**
## What's the purpose of this pull request?
- [x] Moves `ProductCard` component to @faststore/components 
- [x] Adds `ProductCard` styles on @faststore/ui
- [x] Review `ProductCard` structure

## How to test it?
run `yarn dev` and check the `ProductCard`  component or test it using the doc preview link: [ProductCard Component](https://faststore-site-git-feat-product-card-fs-766-alt-faststore.vercel.app/components/molecules/product-card)

### Starter PR
https://github.com/vtex-sites/nextjs.store/pull/368

### CODE REVIEW

**Review by emersonlaurentino — APPROVED** [review]
Is this button supposed to be like this?

Screenshot 2023-02-02 at 10 46 17

**Review by victorhmp — APPROVED** [review]
LGTM

File: apps/site/pages/components/molecules/product-card.mdx [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+    token="--fs-product-card-wide-content-padding"
+    value="var(--fs-spacing-2)"
+  />
+  <TokenRow
+    token="--fs-product-card-wide-bkg-color"
+    value="var(--fs-color-neutral-bkg)"
+    isColor
+  />
+  <TokenRow token="--fs-product-card-wide-min-width" value="9rem" />
+</TokenTable>
```
Comment (hellofanny) [inline apps/site/pages/components/molecules/product-card.mdx:675]:
```suggestion
  <TokenRow token="--fs-product-card-wide-min-width" value="9rem" />
  <TokenRow
    token="--fs-product-card-wide-bkg-color"
    value="var(--fs-color-neutral-bkg)"
    isColor
  />
</TokenTable>
```

File: apps/site/pages/components/molecules/product-card.mdx [hunk 2]
```diff
@@ -0,0 +1,707 @@
...
+
+`data-fs-product-card-actions`
+
+`data-fs-product-card-image`
+
+`data-fs-product-card-heading`
+
+`data-fs-product-card-prices`
+
+`data-fs-product-card-title`
```
Comment (hellofanny) [inline apps/site/pages/components/molecules/product-card.mdx:707]:
```suggestion
`data-fs-product-card-image`

`data-fs-product-card-content`

`data-fs-product-card-heading`

`data-fs-product-card-title`

`data-fs-product-card-prices`

`data-fs-product-card-actions`

```

File: apps/site/pages/components/molecules/product-card.mdx [hunk 3]
```diff
@@ -0,0 +1,707 @@
...
+
+<Tabs items={['Example', 'Code']} defaultIndex="0">
+  <Tab>
+    <OverviewSection
+      containerStyle={{ display: 'grid', gridTemplateColumns: '1fr 1fr 1fr' }}
+    >
+      <ProductCard outOfStock={outOfStock}>
+        <ProductCardImage>
+          <img
+            data-fs-image
```
Comment (hellofanny) [inline apps/site/pages/components/molecules/product-card.mdx:448]: Is this `data-fs-image` here for a specific purpose?

**Review by hellofanny — APPROVED** [review]
LGTM 🚀

File: packages/components/src/molecules/ProductCard/ProductCardImage.tsx [hunk 4]
```diff
@@ -6,18 +6,32 @@ export interface ProductCardImageProps extends HTMLAttributes<HTMLDivElement> {
...
   ) {
     return (
       <div
         ref={ref}
-        data-product-card-image
+        data-fs-product-card-image
         data-testid={testId}
+        style={
+          {
+            '--fs-product-card-image-aspect-ratio': aspectRatio
```
Comment (eduardoformiga) [inline packages/components/src/molecules/ProductCard/ProductCardImage.tsx:32]: interesting trick!

**Review by eduardoformiga — APPROVED** [review]

**Pull request merged**